proxy-node-shared/src/main/java/uk/gov/ida/notification/shared/ProxyNodeLogger.java

```diff
@@ -26,19 +55,15 @@ public void addContext(Exception exception) {
         addContext(ProxyNodeMDCKey.EXCEPTION_STACKTRACE, ExceptionUtils.getStackTrace(exception));
     }
 
-    public void log(Level level, String message) {
-        logWithCallingFrame(level, () -> message);
-    }
-
-    public void log(Level level, Supplier<String> message) {
-        logWithCallingFrame(level, message);
-    }
-
-    private void logWithCallingFrame(Level level, Supplier<String> message) {
+    private void logWithCallingFrame(Level level, Supplier<String> message, Exception exception) {
         Optional<StackWalker.StackFrame> callingFrame = getCallingStackFrame();
-         callingFrame.ifPresent(f -> {
+        callingFrame.ifPresent(f -> {
             addContext(ProxyNodeMDCKey.LOG_LOCATION, String.format("%s.%s", f.getClassName(), f.getMethodName()));
-            LOG.log(level, message);
+            if (exception != null) {
+                LOG.log(level, message.get(), exception);
+            } else {
+                LOG.log(level, message);
+            }
```

Comment:
If we're cleaning up the MDC in the filter, can we remove the line below?

Reply:
No, because this key has to be specific for each log event. The others keys are specific to the whole request.